### ISSUE TITLE
tagOverrideConfig parameter not in effect

### ISSUE THREAD

**Issue by sunny19930321** [apache/incubator-pinot#4891]
`You can also add a tagOverrideConfig section under the tenants section. Currently, the only override allowed is to add additional tags for consuming and completed segments. For example:

"tenants": {
  "broker": "brokerTenantName",
  "server": "serverTenantName",
  "tagOverrideConfig" : {
    "realtimeConsuming" : "serverTenantName_REALTIME"
    "realtimeCompleted" : "serverTenantName_OFFLINE"
  }
}
In the above example, the consuming segments will stil be assigned to serverTenantName_REALTIME hosts, but once they are completed, the segments will be moved to serverTeantnName_OFFLINE. It is possible to specify the full name of any tag in this section (so, for example, you could decide that completed segments for this table should be in pinot servers tagged as allTables_COMPLETED).`

**Comment by npawar** [comment]
Why do you say it is not in effect? We are using this config in our setup, and it is working fine. Consuming segments begin on serverOne_REALTIME and then are periodically relocated to serverOne_OFFLINE by the ControllerPeriodicTask RelocationManager.
What behavior are you seeing?

**Comment by sunny19930321** [comment]
> Why do you say it is not in effect? We are using this config in our setup, and it is working fine. Consuming segments begin on serverOne_REALTIME and then are periodically relocated to serverOne_OFFLINE by the ControllerPeriodicTask RelocationManager.
> What behavior are you seeing?

Thank you for your answer，Through the above table configuration, it is found that the generated segment is still on the original consuming machine and not moved to the serverOne_OFFLINE machine

**Comment by Jackie-Jiang** [comment]
@sunny19930321 In order to move the segment to the servers with another tag, you need to enable the RealtimeSegmentRelocator (a controller periodic task) which is enabled by default. Also, you need to have servers tagged with serverOne_OFFLINE

**Comment by sunny19930321** [comment]
> @sunny19930321 In order to move the segment to the servers with another tag, you need to enable the RealtimeSegmentRelocator (a controller periodic task) which is enabled by default. Also, you need to have servers tagged with serverOne_OFFLINE

yes, the cluster has servers tagged with serverOne_OFFLINE，as for this parameter, RealtimeSegmentRelocator is the default, and my pinot version is from last year

**Comment by Jackie-Jiang** [comment]
> > @sunny19930321 In order to move the segment to the servers with another tag, you need to enable the RealtimeSegmentRelocator (a controller periodic task) which is enabled by default. Also, you need to have servers tagged with serverOne_OFFLINE
> 
> yes, the cluster has servers tagged with serverOne_OFFLINE，as for this parameter, RealtimeSegmentRelocator is the default, and my pinot version is from last year

We cannot trace back to a version one year ago... Is it possible for you to try the latest pinot release - Pinot Apache release 0.2.0?

**Comment by Jackie-Jiang** [comment]
@sunny19930321 Besides, your table only have 1 replica, so relocating can cause downtime (maybe the reason why it does not happen). Can you also try increasing the replication of the table?

**Comment by xiangfu0** [comment]
Closing — this has been implemented in PR #4859. TagNameUtils enhanced to handle TagOverrideConfig, with follow-up fixes in PR #12233 and #12311.
